Highway.cpp

```diff
@@ -18,22 +18,47 @@ void Highway::changeSpeed(int newSpeed)
 
 void Highway::addVehicleInternal(Vehicle* v)
 {
-    assert(false);
+    //assert(false);
 
     /*
     depending on the derived type, call the member function that doesn't evade the cops. 
     */
+    if(auto* vehicle = dynamic_cast<Car*>(v))
+    {
+        vehicle->closeWindows();
+    }
+    if (auto* vehicle = dynamic_cast<Motorcycle*>(v))
+    {
+        vehicle->setSpeed(60);
+    }
+    if (auto* vehicle = dynamic_cast<SemiTruck*>(v))
+    {
+        vehicle->stop();
+    }
```

Comment:
- [ ] use `else if`.  if one cast succeeds you should not continue trying to cast.
- [ ] Motorcycle: Call the other member function that doesn't evade the cops.
- [ ] You need to write a `SemiTruck` member function that does something other than `setSpeed(0)`.  What you're calling here gets called in `removeVehicleInternal()`.  See instruction 3 in main.cpp